opendoas: add package

paper42 opened the pull request:
Maintainer: me
Compile tested: aarch64, Turris MOX, OpenWrt master
Run tested: aarch64, Turris MOX, OpenWrt master

Description: opendoas is a port of OpenBSD doas to Linux, it replaced sudo in the main alpine repository with 3.15.

if someone wants to test this, add this to `/etc/doas.conf`:
```
permit nopass <user> as root
```
and try running doas id -u as `<user>`

paper42 commented:
Should this package go to utils/ or admin/?

neheb commented:
utils probably.

If this is a sudo replacement, wouldn't it make sense to add a symlink?

paper42 commented:
> utils probably.

`sudo` is in admin which makes me a bit confused.

> If this is a sudo replacement, wouldn't it make sense to add a symlink?

It doesn't accept the same arguments, it's not just a sudo rewrite. Alpine has a wrapper script that parses sudo options and translates them to doas, but I don't think that should be added to this package. On OpenBSD and Void, many people don't use any wrappers and just write `doas <something>` or `alias sudo=doas`.

neheb commented:
wrapper script sounds interesting. I assume this can be handled with ALTERNATIVES.

I don't have a strong opinion on this though., I don't use non root users or sudo on OpenWrt.

paper42 commented:
I added a package for the doas - sudo compatibility wrapper that alpine uses, but I am not sure how to use alternatives to avoid conflicts. Is there some documentation I missed?

I converted this PR to a draft for now because it's currently not in a mergeable state.

paper42 commented:
I think I now know how alternatives work, but if we wanted to have doas-sudo-shim to replace sudo, /usr/bin/sudo would have to be a symlink to the real sudo. That sounds wrong and confusing to users to me, what do you think?
```
# opkg install ./doas-sudo-shim_0.1.1_arm_cortex-a9_vfpv3-d16.ipk 
Installing doas-sudo-shim (0.1.1) to root...
Configuring doas-sudo-shim.
Collected errors:
 * pkg_alternatives_update_path: /usr/bin/sudo exists but is not a symlink
```

paper42 commented:
I dropped doas-sudo-shim from this PR

paper42 commented:
I don't think the aarch64 CI failure is caused by this PR. I have been using this package on multiple routers since I created this PR and it seems to work fine.

neheb commented:
https://downloads.openwrt.org/snapshots/faillogs/x86_64/packages/opendoas/compile.txt

paper42 commented:
> [downloads.openwrt.org/snapshots/faillogs/x86_64/packages/opendoas/compile.txt](https://downloads.openwrt.org/snapshots/faillogs/x86_64/packages/opendoas/compile.txt)

hopefully fixed in #19183
